modules/03_in_memory_storage/README.md

```diff
@@ -1,9 +1,471 @@
+
+To do this, we'll need:
+1. a text field on the web page with a button "submit" (this will send data to the server)
+2. an endpoint on the server that would accept this data.
+
+#### 1. Create HTML page for word upload
+
+Let's create a web page that will contain a text field for the word to upload and a button `submit`.
+
+Create a new file `upload_word.html` in the folder `templates`.
+Inherit it from `base.html`, like we did it in the previous stage, and put the following form inside `block content`:
+
+```html
+<form action = "/upload_word" method = "POST" enctype="multipart/form-data">
+    <div class="form-group">
+        <label for="secretId">Secret</label>
+        <input type="text" class="form-control" id="secretId" name="secretWord" aria-describedby="secretHelp" placeholder="Secret Word">
+        <small id="secretHelp" class="form-text text-muted"></small>
+    </div>
+
+	<button type="submit" class="btn btn-primary">Submit</button>
+</form>
+```
+
+#### 2. Serve this HTML page on HTTP endpoint
+
+To let users open this page, add a new endpoint in `api.py`:
+
+```python
+@app.route('/words', methods=['GET'])
+def upload_word_page():
+    return render_template('words.html')
+```
+
+Now start the app, and verify that you can see this form on the endpoint `/words`.
+
+> Most likely, the app will start on localhost:5000. In this case you can use this link to open the page:
+> [http://localhost:5000/words](http://locahost:5000)
+
+
+When you type in a word in the form and click "submit", your browser will try to
+send it to the endpoint `/upload_word`. This is the endpoint we specified in the
+`action` attribute of the form.
+
+Because this endpoint doesn't exist yet, in the browser you'll see a "Not Found" page,
+and in the logs of the running application you'll see the following entry:
+
+```
+127.0.0.1 - - [01/Jan/2022 11:12:34] "POST /upload_word HTTP/1.1" 404 -
+```
+
+#### 3. Accept form data on the server
+
+Let's create the endpoint `/upload_word` that will actually accept the form data.
+
+```python
+from flask import redirect, request
+
+@app.route('/upload_word', methods=['POST'])  # browsers send POST request when submitting a form
+def upload_word():
+    # request.form is a special variable in Flask that will contain the form data
+    secret_word = request.form['secretWord']  # note the "name" attribute of the <input> we have in HTML
+    print("Uploaded word " + repr(secret_word))
+    return redirect('/')  # redirect back to the main page
+```
+
+Run the application, try uploading a word, and verify that the upload works now. 
+Verify that in the application log you see the uploaded word. 
+
+#### 4. Give user feedback
+
+It's good to give the user know too, that the word was uploaded.
+
+Flask has a concept of flashing that we can use. In the logic of our endpoint we 
+would "flash" one or more messages, and these messages will become available 
+to us when rendering a template that we show to the user.
```

Comment:
Linkebreaks are useful so that you don't scroll too much when we read in non preview mode.

Reply:
Well, if you two insist. For me, who uses a really wide monitor, I prefer to choose where my paragraphs end, instead of having them predefined in the markdown. On my monitor the markdown file only covers half the file that's open 😸